gae/js/app/views/rate_after_date.js

```diff
@@ -0,0 +1,149 @@
+define(
+    [
+        'backbone', 'backform', 'bootstrap','bootstrap-datepicker',
+        'app/models/rate_after_date',
+        'text!app/templates/staffposition_rates.html'
+    ],
+    function(Backbone, Backform, bootstrap, bsdp,
+              NewRateAfterDate,
+             template) {
+        var fields = [
+            {
+                name: "rate",
+                label: "New Rate",
+                required: true,
+                control: "input",
+            },
+            {
+                id: "datepicker",
+                name: "date",
+                label: "Start Date",
+                required: true,
+                control: "datepicker",
+                options: {autoclose: true, assumeNearbyYear: true, todayHighlight: true, format: "yyyy-mm-dd"},
+            },
+            {
+                name: "type",
+                label: "Rate Type",
+                control: "select",
+                required: true,
+                options: [
+                    {label: "--- select rate type---", value: ""},
+                    {label: "Hourly", value: "hourly_rates"},
+                    {label: "Mileage", value: "mileage_rates"},
+                ]
+            },
+            {
+                control: "button",
+                extraClasses: ['btn-primary', 'btn-sm'],
+                name: "addRate",
+                label: "Add Rate"
+            }
+        ];
+        var View = Backbone.View.extend({
+            el: '#staffposition-form-view',
+            events: {
+                'click button.remove-hourly-rate': 'removeHourlyRate',
+                'click button.remove-mileage-rate': 'removeMileageRate',
+                'click button.edit-hourly-rate': 'editHourlyRate',
+                'click button.edit-mileage-rate': 'editMileageRate',
+                'click button[name=addRate]': 'addRate',
+            },
+            initialize: function(options) {
+                this.options = options;
+                this.template = _.template(template);
+                this.model = new NewRateAfterDate({id: this.options.staffposition.get('id')});
+                this.hourly_rates = this.options.staffposition.get('hourly_rates');
```

Comment:
These could be done as this.rates['hourly'] ...
And then you could reuse some code below. But this is not bad.